src/avt/Filters/avtResampleFilter.C

```diff
@@ -561,15 +570,9 @@ avtResampleFilter::ResampleInput(void)
                 if (vars[i]->GetNumberOfComponents() == 1)
                 {
                     if (cellCenteredOutput)
-                    {
-                        rg->GetCellData()->AddArray(vars[i]);
                         rg->GetCellData()->SetScalars(vars[i]);
-                    }
                     else
-                    {
-                        rg->GetPointData()->AddArray(vars[i]);
                         rg->GetPointData()->SetScalars(vars[i]);
```

Comment:
The original code had both an AddArray and a SetScalar. The AddArray is redundant as SetScalars does that.

int vtkDataSetAttributes::SetAttribute(vtkAbstractArray* aa, int attributeType)
// This method lets the user add an array and make it the current
// scalars, vectors etc... (this is determined by the attribute type
// which is an enum defined vtkDataSetAttributes)

That said, in this step arrays are being added to a new vtkRecGrid. If you look at the rest of the vector sniper SetVectors and SetTensors are used without the AddArray.

Reply:
The VTK documentation for the [SetAttribute method](https://vtk.org/doc/nightly/html/classvtkDataSetAttributes.html#afac42658dd14584d312f132556f83a96) specifically states that any array previously set for that attribute type will be removed from the DataSetAttributes.

The for-loop indicates it could possibly be adding multiple scalars/vectors/tensors. In which case, only the last added Scalar array will still be in the dataset attributes at then end of the for loop.  Am I misunderstanding the documentation? 

The original logic was probably wrong as well. It should be 'AddArray', followed by SetActiveScalars.
Actually, not sure the SetActiveScalars/Tensors/Vectors is even needed. Most of our filters operate on named arrays and only set the active attributes as needed.